src/core/data-include.js

```diff
@@ -14,28 +14,76 @@ import { runTransforms } from "./utils.js";
 
 export const name = "core/data-include";
 
+/**
+ * Calculates indentation when the element starts after a newline.
+ * The value will be empty if no newline or any non-whitespace exists after one.
+ * @param {Element} element
+ */
+function getElementIndentation(element) {
+  const { previousSibling } = element;
+  if (!previousSibling || previousSibling.nodeType !== Node.TEXT_NODE) {
+    return "";
+  }
+  const index = previousSibling.textContent.lastIndexOf("\n");
+  if (index === -1) {
+    return "";
+  }
+  const slice = previousSibling.textContent.slice(index + 1);
+  if (/[^\s]/.test(slice)) {
+    return "";
+  }
+  return slice;
+}
+
+/**
+ * @param {string} text
+ * @param {string} indent
+ */
+function indentTextWithoutFirstLine(text, indent) {
+  const lines = text.split("\n");
+  const firstLine = lines.shift();
+  return `${firstLine}\n${lines.map(line => indent + line).join("\n")}`;
+}
+
+/**
+ * @param {HTMLElement} el
+ * @param {string} data
+ * @param {boolean} replace
```

Comment:
I kinda like it that TypeScript makes boolean traps less bad. Stylistically tho, I wonder if we should still use a property bag here?